File: cppFinal/IceCreamCup.cpp
#include "IceCreamCup.h"
#include <iostream>
using namespace std;

IceCreamCup::IceCreamCup(string flavor_, string type_) 
: type(type_), IceCream(flavor_) {

}

void IceCreamCup::addTopping(string topping) {
    if (topping.compare("bonnet") == 0) {
        cout << "sorry msg - no bonnet" << endl;
    }
    else {
        IceCream::addTopping(topping);
    }
}

void IceCreamCup::display() {
    cout << "This ice cream is in a " << type << endl;
    IceCream::display();
}

IceCreamCup::~IceCreamCup() {

}

File: javaFinal/src/javaFinal/IceCreamCup.java
package javaFinal;

public class IceCreamCup extends IceCream implements iIceCream {

	private String type; //cup or cone
	
	public IceCreamCup(String flavor, String type) {
		super(flavor);
		this.type = type;
	}
	
	@Override
	public void addTopping(String topping) {
		if (topping.equals("bonnet")) {
			System.out.println("sorry msg - no bonnet");
		}
		else {
			super.addTopping(topping);
		}
	}
	
	@Override
	public void display() {
		System.out.println("This ice cream is in a " + type);
		super.display();
	}

}
